crates/c-api/include/wasmtime/profiling.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * \file wasmtime/profiling.h
+ *
+ * \brief API for Wasmtime guest profiler
+ */
+
+#ifndef WASMTIME_PROFILING_H
+#define WASMTIME_PROFILING_H
+
+#include <wasm.h>
+#include <wasmtime/error.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct wasmtime_guestprofiler_t wasmtime_guestprofiler_t;
+
+WASM_API_EXTERN void wasmtime_guestprofiler_delete(
```

Comment:
Mind adding some documentation for this and the functions below? One thing I think is especially important to document is the ownership of pointers and who's responsible for deallocating what, for example.